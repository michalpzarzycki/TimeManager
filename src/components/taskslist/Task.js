import React from 'react'
import styles from './Task.module.css'
export default function Task() {
    return <div className={styles.tasksInfoBarContainer}>
    <div className={styles.checkbox}>
        <input type="checkbox"/>
    </div>
    <div className={styles.id}>ID</div>
    <div className={styles.avatar}></div>
    <div className={styles. title}>TYTUL</div>
    <div className={styles.importance}>WAZNOSC</div>
    <div className={styles.deadline}>{Date.now()}</div>
    <div className={styles.buttons}>
        <div className={styles.details}></div>
        <div className={styles.edit}></div>
        <div className={styles.delete}></div>
    </div>
</div>
}