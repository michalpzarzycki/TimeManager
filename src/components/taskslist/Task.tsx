import React, { useState } from 'react'
import styles from './Task.module.css'
import TaskEditPopup from './TaskEditPopup';
import TaskEditButtons from './TaskEditButtons';
import TaskDeadline from './TaskDeadline';
import TaskImportance from './TaskImportace';
import TaskCheckBox from './TaskCheckBox';
import TaskTitle from './TaskTitle';
import TaskDone from './TaskDone';
import TaskLoader from './TaskLoader';
import route1Service from '../../services/route1Service';
import { connect } from 'react-redux';
import {DARK_ROUTE1, LIGHT_ROUTE1} from '../../variables'
const { darkBackground, darkColor} = DARK_ROUTE1;
const { lightBackground, lightColor} = LIGHT_ROUTE1

function Task({ task, handleDeletePopup, handleDetailsPopup, darkMode }: any) {
    const [popup, setPopup] = useState(false);
    const [editedTask, setEditedTask] = useState("")
    const [editedDeadline, setEditedDeadline] = useState("")
    const [editedImportance, setEditedImportance] = useState("")
    const [editedPhoto, setEditedPhoto] = useState("")
    const [isEdited, setIsEdited] = useState(false)
    const [isLoading, setIsLoading] = useState(false)


    const handleEdit = () => setPopup(true)

    function handleChange(event: any) {
        if (event.target.name === 'task') setEditedTask(event.target.value)
        if (event.target.name === 'importance') setEditedImportance(event.target.value)
        if (event.target.name === 'deadline') setEditedDeadline(event.target.value)
        if (event.target.name === 'photo') setEditedPhoto(event.target.value)
    }
    function handleSubmit(event: any) {
        event.preventDefault()
        setIsLoading(true)
        route1Service.editTask(task.docId, { task: editedTask, deadline: editedDeadline, importance: editedImportance, photo: editedPhoto})
                     .then(() => {
                            setTimeout(() => {
                                setIsLoading(false)
                            }, 1000)
                            setTimeout(() => {
                                setIsEdited(false)
                            }, 6000)
                            setIsEdited(true) })  
                     .catch(() => {
                            setTimeout(() => {
                                setIsLoading(false)
                            }, 1000)
                            setTimeout(() => {
                                setIsEdited(false)
                            }, 6000)
        })

    }
    const handleChecked = (task: any) => route1Service.changeDoneTask(task.docId, task.done)
    
    const setDoneTask = (task: any) => route1Service.setTaskCompleted({...task, doneDate: Date.now()}, task.docId)
    
    const setUncompletedTask = (task: any) => route1Service.setTaskUncompleted({...task, uncompletedDate: Date.now()}, task.docId)

    return <div className={styles.tasksInfoBarContainer} style={{backgroundColor: darkMode ? darkBackground : lightBackground, borderColor: darkMode ? darkColor : lightColor}}>
                <TaskLoader 
            j         isLoading={isLoading} />
                <TaskEditPopup
                    handleChange={handleChange}
                    isEdited={isEdited}
                    popup={popup}
                    setPopup={setPopup}
                    handleSubmit={handleSubmit}/>
                <TaskCheckBox 
                    task={task} 
                    handleChecked={handleChecked} />
                <TaskDone  
                    task={task} 
                    setDoneTask={setDoneTask} 
                    setUncompletedTask={setUncompletedTask} />
                <TaskTitle 
                    task={task} />
                <TaskImportance 
                    task={task} />
                <TaskDeadline 
                    task={task} />
                <TaskEditButtons
                    handleDeletePopup={handleDeletePopup}
                    handleDetailsPopup={handleDetailsPopup}
                    handleEdit={handleEdit}
                    task={task} />
        </div>
}
const mapStateToProps = (state: any) => {
    return {
        darkMode: state.darkmode
    }
}
export default connect(mapStateToProps)(Task)