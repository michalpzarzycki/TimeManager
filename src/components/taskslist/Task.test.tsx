import React from 'react'
import {shallow} from 'enzyme'
import {Provider} from 'react-redux'
import configureMockStore from "redux-mock-store";
import Task from './Task';
const mockStore = configureMockStore();
const store = mockStore({});

test('Expext to render TASK component', () => {
    expect(shallow(<Provider store={store}><Task /></Provider>)).toMatchSnapshot()
})