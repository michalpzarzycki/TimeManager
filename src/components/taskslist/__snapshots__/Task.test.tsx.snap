// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expext to render TASK component 1`] = `ShallowWrapper {}`;
